genesis/public/core/instance_test.go

```diff
@@ -0,0 +1,267 @@
+	_, err := newInstanceDomainProxy(nil)
+	assert.EqualError(t, err, "parent must not be nil")
+}
+
+func TestInstanceDomainProxy_CreateInstance(t *testing.T) {
+	parent := &mockParent{}
+	proxy, err := newInstanceDomainProxy(parent)
+	assert.NoError(t, err)
+
+	factory := &mockFactory{}
+	registered, err := proxy.CreateInstance(factory)
+	assert.NoError(t, err)
+
+	_, err = uuid.FromString(registered)
+	assert.NoError(t, err)
+}
+
+func TestInstanceDomainProxy_CreateInstance_WithError(t *testing.T) {
+	parent := &mockParent{}
+	proxy, err := newInstanceDomainProxy(parent)
+	assert.NoError(t, err)
+
+	factory := &mockFactoryError{}
+	_, err = proxy.CreateInstance(factory)
+	assert.EqualError(t, err, "factory returns nil")
+}
+
+func TestInstanceDomainProxy_GetInstance(t *testing.T) {
+	parent := &mockParent{}
+	proxy, err := newInstanceDomainProxy(parent)
+	assert.NoError(t, err)
+
+	factory := &mockFactory{}
+	registered, err := proxy.CreateInstance(factory)
+	assert.NoError(t, err)
+
+	resolved, err := proxy.GetInstance(registered)
+	assert.NoError(t, err)
+
+	assert.Equal(t, &mockProxy{
+		parent: proxy.instance,
+	}, resolved)
+}
+
+func TestInstanceDomainProxy_GetReference(t *testing.T) {
+	parent := &mockParent{}
+	proxy, err := newInstanceDomainProxy(parent)
+	assert.NoError(t, err)
+
+	reference := proxy.GetReference()
+	// TODO should return actual reference
+	assert.Nil(t, reference)
+}
+
+func TestInstanceDomainProxy_GetInstance_IncorrectRef(t *testing.T) {
+	parent := &mockParent{}
+	proxy, err := newInstanceDomainProxy(parent)
+	assert.NoError(t, err)
+
+	_, err = proxy.GetInstance("1")
+	assert.EqualError(t, err, "object with record 1 does not exist")
+}
+
+func TestInstanceDomainProxy_GetParent(t *testing.T) {
+	parent := &mockParent{}
+	refDomainProxy := newReferenceDomainProxy(parent)
+
+	returnedParent := refDomainProxy.GetParent()
+	assert.Equal(t, parent, returnedParent)
+}
+
+func TestInstanceDomainProxy_GetClassID(t *testing.T) {
+	parent := &mockParent{}
+	proxy, err := newInstanceDomainProxy(parent)
+	assert.NoError(t, err)
+
+	assert.Equal(t, class.InstanceDomainID, proxy.GetClassID())
+}
+
+func TestNewInstanceDomainFactory(t *testing.T) {
+	factory := NewInstanceDomainFactory()
+	assert.Equal(t, &instanceDomainFactory{}, factory)
+}
+
+func TestInstanceDomainFactory_GetClassID(t *testing.T) {
+	factory := NewInstanceDomainFactory()
+	assert.Equal(t, class.InstanceDomainID, factory.GetClassID())
+}
+
+func TestInstanceDomainFactory_GetReference(t *testing.T) {
+	factory := NewInstanceDomainFactory()
+	assert.Nil(t, factory.GetReference())
+}
+
+func TestInstanceDomainFactory_Create(t *testing.T) {
+	parent := &mockParent{}
+	factory := NewInstanceDomainFactory()
+	proxy := factory.Create(parent)
+	instDomain, _ := newInstanceDomain(parent)
+
+	assert.Equal(t, &instanceDomainProxy{
+		instance: instDomain,
+	}, proxy)
+}
+
+func TestInstanceDomainFactory_CreateWithError(t *testing.T) {
```

Comment:
There need to be two tests with errors:  
- with no parent
- with parent, whose AddChild returns error